### PR TITLE
tests: Fix CLI unit tests

### PR DESCRIPTION
**Pull request by michaljurecko**
Jira: https://keboola.atlassian.net/browse/PSGO-447

**Changes:**
- Fixed bugs in CLI unit tests (from the new initiative), and other minor fixes.

-----------

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -59,21 +59,21 @@ require (
 	go.etcd.io/etcd/tests/v3 v3.5.11
 	go.nhat.io/aferocopy/v2 v2.0.1
 	go.opencensus.io v0.24.0
-	go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc v0.46.0
+	go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc v0.48.0
 	go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.44.0
```
Comment (michaljurecko) [inline go.mod:63]: Updated lib, to fix data race, but may be we have to wait for `0.48.1`:
https://github.com/open-telemetry/opentelemetry-go-contrib/issues/4895

File: internal/pkg/service/stream/storage/level/local/writer/volume/file_test.go [hunk 2]
```diff
@@ -41,8 +41,9 @@ func (f *testFile) Sync() error {
 }
 
 func (f *testFile) Close() error {
+	err := f.file.Close()
 	if f.CloseError != nil {
 		return f.CloseError
 	}
-	return f.file.Close()
+	return err
 }
```
Comment (michaljurecko) [inline internal/pkg/service/stream/storage/level/local/writer/volume/file_test.go:49]: Fix one more test on Windows, always close the underlying file.

File: internal/pkg/service/cli/dialog/init_test.go [hunk 3]
```diff
@@ -109,10 +109,9 @@ func TestDialogs_AskInitOptions_No_CI(t *testing.T) {
 	o.Set("ci", "false")
 	o.Set("branches", "main")
 
-	f := syncInit.Flags{
-		Branches: configmap.NewValue("main"),
-		CI:       configmap.NewValue(false),
-	}
+	f := syncInit.DefaultFlags()
+	f.Branches = configmap.NewValueWithOrigin("main", configmap.SetByFlag)
+	f.CI = configmap.NewValueWithOrigin(false, configmap.SetByFlag)
```
[the remark this answers is not in the document]
Comment (hosekpeter) [reply internal/pkg/service/cli/dialog/init_test.go:114]: image

Yes, I found this out in several tests

**Review by hosekpeter — APPROVED** [review]
LGTM

**Pull request merged**